test/Governance.test.ts

```diff
@@ -0,0 +1,576 @@
+    expect(await timelock.getMinDelay()).to.eq(newMinDelay);
+    expect(await locking.minCliffPeriod()).to.eq(newMinCliff);
+    expect(await locking.minSlopePeriod()).to.eq(newMinSlope);
+  });
+
+  it('should change governor roles', async function () {
+    const proposerRole = await timelock.PROPOSER_ROLE();
+    const cancellerRole = await timelock.CANCELLER_ROLE();
+
+    const newProposer = alice.address;
+    const newCanceller = charlie.address;
+    const watchdogAddress = await governanceFactory.watchdogMultiSig();
+
+    const targets = [
+      governanceAddresses.TimelockController,
+      governanceAddresses.TimelockController,
+      governanceAddresses.TimelockController,
+      governanceAddresses.TimelockController,
+    ];
+    const values = Array(4).fill(0);
+    const calldatas = [];
+    const description = 'Change governor roles';
+
+    calldatas.push(
+      timelock.interface.encodeFunctionData('grantRole', [
+        proposerRole,
+        newProposer,
+      ]),
+    );
+
+    calldatas.push(
+      timelock.interface.encodeFunctionData('grantRole', [
+        cancellerRole,
+        newCanceller,
+      ]),
+    );
+    calldatas.push(
+      timelock.interface.encodeFunctionData('revokeRole', [
+        proposerRole,
+        governanceAddresses.MentoGovernor,
+      ]),
+    );
+    calldatas.push(
+      timelock.interface.encodeFunctionData('revokeRole', [
+        cancellerRole,
+        watchdogAddress,
+      ]),
+    );
+
+    // Create a proposal to change governor roles
+    const proposalId = submitProposal(
+      alice,
+      targets,
+      values,
+      calldatas,
+      description,
+    );
+
+    // Vote on the proposal using multiple accounts, with the majority voting YES
+    await governor.connect(alice).castVote(proposalId, 0);
+    await governor.connect(bob).castVote(proposalId, 1);
+    await governor.connect(charlie).castVote(proposalId, 1);
+
+    // Voting period
+    await timeTravel(7);
+
+    await governor.connect(alice)['queue(uint256)'](proposalId);
+
+    // Timelock period
+    await timeTravel(2);
+
+    expect(await timelock.hasRole(proposerRole, newProposer)).to.be.false;
+    expect(await timelock.hasRole(cancellerRole, newCanceller)).to.be.false;
+    expect(
+      await timelock.hasRole(proposerRole, governanceAddresses.MentoGovernor),
+    ).to.be.true;
+    expect(await timelock.hasRole(cancellerRole, watchdogAddress)).to.be.true;
+
+    await governor.connect(alice)['execute(uint256)'](proposalId);
+
+    expect(await timelock.hasRole(proposerRole, newProposer)).to.be.true;
+    expect(await timelock.hasRole(cancellerRole, newCanceller)).to.be.true;
+    expect(
+      await timelock.hasRole(proposerRole, governanceAddresses.MentoGovernor),
+    ).to.be.false;
+    expect(await timelock.hasRole(cancellerRole, watchdogAddress)).to.be.false;
+  });
+
+  it('should upgrade upgradable contracts', async function () {
```

Comment:
just a note that we'll need to include the `Emissions` contract in this test as well after https://github.com/mento-protocol/mento-core/pull/365 is merged

leave it up to you if you want to do it in this PR or create a new issue